pkg/cache/internal/cache_reader.go

```diff
@@ -106,14 +108,24 @@ func (c *CacheReader) Get(_ context.Context, key client.ObjectKey, out client.Ob
 }
 
 // List lists items out of the indexer and writes them to out.
-func (c *CacheReader) List(_ context.Context, out client.ObjectList, opts ...client.ListOption) error {
+func (c *CacheReader) List(ctx context.Context, out client.ObjectList, opts ...client.ListOption) error {
 	var objs []interface{}
 	var err error
 
 	listOpts := client.ListOptions{}
 	listOpts.ApplyOptions(opts)
 
+	// TODO(kcp), should we just require people to pass in the cluster list option, or maybe provide
+	// a wrapper that adds it from the context automatically rather than doing this?
+	// It may also make more sense to just use the context and not bother provided a ListOption for it
```

Comment:
I maybe think I prefer the ListOption, especially if it's in a kcp repo instead of controller-runtime. I'm not sure that would work with the live client easily, though, as that needs the context?